CAFAna/Systs/RecoEnergyNDSysts.h

```diff
@@ -0,0 +1,517 @@
+#pragma once
+
+#include "CAFAna/Core/ISyst.h"
+#include "CAFAna/Core/Utilities.h"
+#include "CAFAna/Cuts/AnaCuts.h"
+
+#include "StandardRecord/SRProxy.h"
+
+#include "TFile.h"
+#include "TH1.h"
+#include "TRandom3.h"
+
+#include <cassert>
+#include <string>
+#include <vector>
+#include <algorithm>
+
+// Implement ND energy scale uncertainties on truth information
+// for use in eRecProxy
+// LepE, eP, ePip, ePim, ePi0, eother
+class TH1;
+class TH2;
+
+namespace ana
+{
+  class SystShifts;
+
+  class RecoEnergyScaleND : public ISyst {
+  public:
+    RecoEnergyScaleND() : ISyst("RecoEnergyScaleND", "Global Reco Energy Scale ND Syst") {}
+    void Shift(double sigma,
+               Restorer& restore, 
+               caf::SRProxy* sr,
+               double& weight) const override {
+  
+      restore.Add(sr->EVisReco_ND,
+                  sr->HadEVisReco_ND,
+                  sr->Elep_reco);
+
+      const double scale = 0.02 * sigma;
+      if (!sr->isFD) {
+        // To match FD:
+        // Whether NC or CC, Numu or Nue, we want to shift total "reconstructed neutrino energy"
+        // by the "reconstructed hadronic energy".
+        sr->EVisReco_ND += (sr->EVisReco_ND - sr->Elep_reco) * scale;
+        // Also shift the hadronic energy variable, if we don't do this then shifted 
+        // plots of the reconstructed hadronic energy will not be different to nominal.
+        sr->HadEVisReco_ND *= 1. + scale;
+        // If it isn't Numu-CC, also shift the "reconstructed neutrino energy" by the
+        // "reconstructed leptonic energy".
+        // Also shift the reconstructed leptonic energy itself.
+        if (!(sr->isCC && abs(sr->nuPDG) == 14)) {
+          sr->EVisReco_ND += sr->Elep_reco * scale; 
+          sr->Elep_reco *= 1. + scale;
+        }
+      }
+    }
+  };
```

Comment:
Implementation code should be in .cxx file

Reply:
I followed how things were done in EnergyScale.h when making this ND version of the energy scale systematics. It can be moved to the .cxx for both.